demo-android/src/main/java/nl/tudelft/ipv8/android/demo/coin/WalletManager.kt

```diff
@@ -87,19 +98,57 @@ class WalletManager(
 
             override fun doneDownload() {
                 super.doneDownload()
-                Log.w("Coin", "Download Complete!")
+                progress = 100
+                Log.i("Coin", "Download Complete!")
                 Log.i("Coin", "Balance: ${kit.wallet().balance}")
                 isDownloading = false
             }
         })
 
         Log.i("Coin", "Coin: starting the setup of kit.")
         kit.setBlockingStartup(false)
-        kit.startAsync()
-        kit.awaitRunning()
+            .startAsync()
+            .awaitRunning()
+
+        if (publicPrivateKeyPair != null) {
```

Comment:
Do we know if this works if we also supply a serializedDeterministicKey, so serializedDeterministicKey + publicPrivateKeyPair?

Reply:
No for know it's either. Probably best to create different contructors for this instead of passing null values

Reply:
The protocolECKey() function though, it returns the public bitcoin key we use in the whole protocol. It is always the same, namely the first key that is issued from the mnemonic seed we use, like this: `return kit.wallet().issuedReceiveKeys[0]`

So say if user A with seed A at one point joined a multi-sig wallet, and he later wants to access it again, they must use seed A again. Importing a public/private key pair is not the protocol key. I think the public/private key pair can be used to pay to multi-sig wallets with though.

Maybe a good idea to make this clear to the user, doesn't have to be right now.